tsconfig.json

```diff
@@ -1,6 +1,8 @@
 {
 	"compilerOptions": {
-    "moduleResolution": "Node",
+      "moduleResolution": "Node16",
+	  "module": "Node16",
+	  "target": "ES2018",
```

Comment:
Was there any specific reasoning for these options?

Reply:
After updating `typescript` and `tslib` to latest, `npx tsc` fails with:

```text-plain
error TS1501: This regular expression flag is only available when targeting 'es6' or later.
```

Setting `compilerOptions.target` to `ES6`, then to `ES2018` (due to a similar error) solves the issue.

However building now fails with:

```text-plain
error TS1202: Import assignment cannot be used when targeting ECMAScript modules. Consider using 'import * as ns from "mod"', 'import {a} from "mod"', 'import d from "mod"', or another module format instead.
```

Setting `compilerOptions.moduleResolution` to `Node16` fails with:

```text-plain
tsconfig.json:2:2 - error TS5110: Option 'module' must be set to 'Node16' when option 'moduleResolution' is set to 'Node16'.
2  "compilerOptions": {
  ~~~~~~~~~~~~~~~~~

Found 1 error in tsconfig.json:2
```

This is fixed by setting `compilerOptions.module` to the same value, as indicated by the error.

This causes some build errors regarding unknown types. Seems that TypeScript with its initial module resolution configuration would find types in the same directory automatically, now they need to be manually exported and imported.

Reply:
I don't know how others see this, but it feels a bit like fighting symptoms but not the problem.
E.g. I wouldn't set `moduleResolution` or `module` to `Node16`.
It won't really break the code since node is backwards compatible, but I think it isn't really right when we use node18 (or 20 in the future ;) )

Reply:
> but I think it isn't really right when we use node18

It's not tying it to a specific version of node, see the explanation here https://www.typescriptlang.org/docs/handbook/modules/reference.html#node16-nodenext-1 and the reason to use `node16` over `nodenext` is even though they are currently equivalent, the `node16` will keep us in the current stable system and `nodenext` will be moved/forked to the new module system changes.

I think potentially a `moduleResolution` of `bundler` and `module` of `esnext` would have worked as well based on what I can gather from the errors listed

Reply:
@alexpietsch , Node16 is more like Node stable.
I choose the Node-specific ones instead of ESNext since we still have the old import syntax. When we convert the server to ESM, then we can get rid of it thoroughly.

Reply:
@rauenzi you are right, my fault.
@eliandoran Yes, not using ESM is what I mean, I also had ts complaining because of this. We should convert it soon, I think.

Reply:
@alexpietsch , we'll do that in #15 . Whether we'll do it soon or not will depend on how complicated it'll be.